### ISSUE TITLE
programmatically sorting does not always update asc/des sort icon

### ISSUE THREAD

**Issue by igorlino** [lorenzofox3/Smart-Table#571]
Hi,

I'm playing around with the smartTableCtrl  in order to set a sort function.

The table has by default a sort function in one of the columns as the values are to be sorted by date.
That all works perfectly. Clicking the column, changes the mode to asc or desc or disable the sorting. All fine.

The thing that is partially working is when I what to change the sort programmatically.

NOT WORKING
smartTableCtrl.sortBy(function sorter(row) {...}, reverse=true)

basically, it sorts the rows, but the icon is not updated with respect to asc or des.

Seems something happening in the stSort.js file.

```
//table state --> view
        scope.$watch(function () {
          return ctrl.tableState().sort;
        }, function (newValue) {
          if (newValue.predicate !== predicate) {
            index = 0;
            element
              .removeClass(classAscent)
              .removeClass(classDescent);
          } else {
            index = newValue.reverse === true ? 2 : 1;
            element
              .removeClass(stateClasses[index % 2])
              .addClass(stateClasses[index - 1]);
          }
        }, true);
```

Doing some breakpoint, shows that the reverse is not 'true'  (although it was true in the sortBy parameter)
and also, in the column that is to be evaluated.

the classes update wrong because:

index = 0
newValue.precate === predicate

therefore,

stateClasses[index - 1] = stateClasses[0 - 1] = stateClasses[-1]  = undefined

so addClass(undefined) of course does not give anything.

WORKING
For calling programmatically  the sortBy without functions, only string will work/sort fine.  i.e.    smartTableCtrl.sortBy('DueDate', true)

**Comment by lorenzofox3** [comment]
can you provide a plunker please ? Because one of the example in the documentation website uses getter functions and I don't see any issue. So I'll need a little bit of context

**Comment by igorlino** [comment]
Hi, took me sometime to get back to my computer.... 

I have now re-created an example. I found the getter example you said, its in the Sort Data example.

So, I forked the smart-table-website project, then add a bottom that programmatically sorts by "First Name"  and Reverse=true
into that example:

https://github.com/igorlino/smart-table-website/commit/039fc8cbbe01cbfdbcb1e17fb23f0d66dd5a98f2


So, when you press the button, you see the table gets sorted but the icon does not get updated.

BTW, it would be kinda cool to have that service directive (from my commit) as part of the smart table, its really nice to use, almost like a swiss knive.
